File: 02_selection_sort/javascript/02_selection_sort.js
'use strict';

function searchLower(arr) {
  let lower = arr[0];
  let lowerIndex = 0;

  for (let i in arr) {
    if(arr[i] < lower) {
      lower = arr[i];
      lowerIndex = i;
    }
  }

  return lowerIndex;
}

function selectionSort(arr) {
  let length = arr.length;
  let newArr = [];

  for(let i = 0; i < length - 1; i++) {
    let lowerIndex = searchLower(arr);
    newArr.push(+arr.splice(lowerIndex, 1).join(""));
  }
    
  return newArr;

}

const list = [5, 6, 8, ,2, 1];


console.log(selectionSort(list));

File: 02_selection_sort/python/02_selection_sort.py
def searchLower(arr):
  lower = arr[0]
  lower_index = 0

  for i in range(1, len(arr)):
    if arr[i] < lower:
      lower = arr[i]
      lower_index = i
  return lower_index


def selectionSort(arr):
  new_arr = []

  for i in range(1, len(arr)):
    lower_index = searchLower(arr)
    new_arr.append(arr.pop(lower_index))

  return new_arr

list = [10,6,4,8,12,16,1,3]


print(selectionSort(list))
